DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Model/ProcessContext.cs

```diff
@@ -0,0 +1,21 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using Dicom;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Model
+{
+    public class ProcessContext
+    {
+        public HashSet<string> VisitedNodes { get; set; } = new HashSet<string>();
```

Comment:
Soft Suggestion:
Better to make this a private member, then expose two methods:
void Visit(string item); //simply calls VisitedNodes.Add()
bool isVisited(string item); //calls VisitedNodes.Contains()